### PR TITLE
Add some scheduler documentation

### PR DESCRIPTION
**Pull request by jpsamaroo**
Including fault tolerance docs that I should've added previously...

### CODE REVIEW

File: src/fault-handler.jl [hunk 1]
```diff
@@ -6,9 +6,27 @@ check_exited_exception(res::RemoteException) =
 check_exited_exception(res::ProcessExitedException) = true
 check_exited_exception(res) = false
 
+"""
+    handle_fault(...)
+
+An internal function to handle a worker dieing or being killed by the OS.
```
Comment (abx78) [inline src/fault-handler.jl]: you might want to use `dying` here as more common rather than `dieing` (very specialized term)

Comment (jpsamaroo) [reply src/fault-handler.jl]: Good catch :slightly_smiling_face:

Comment (abx78) [reply src/fault-handler.jl]: My first review is about dead stuff 💀 sob 🗡

**Comment by jpsamaroo** [comment]
Any Dagger maintainers care to review this? If not I'll merge in a few days.

File: src/fault-handler.jl [hunk 2]
```diff
@@ -6,9 +6,27 @@ check_exited_exception(res::RemoteException) =
...
+to recover from the fault.
+
+Note: The logic for this functionality is not currently perfectly robust to
+all failure modes, and is only really intended as a last-ditch attempt to
+repair and continue executing. While it should never cause incorrect execution
+of DAGs, it *may* cause a `KeyError` or other failures in the scheduler due to
+the complexity of getting the internal state back to a consistent and proper
+state.
+"""
 function handle_fault(ctx, state, thunk, oldproc, chan, node_order)
```
Comment (abx78) [inline src/fault-handler.jl]: Is it maybe worth to mention what the arguments of the handle_func are?

Comment (jpsamaroo) [reply src/fault-handler.jl]: I would, if only I knew the purpose of each argument :smile:

**Pull request merged**